Watchman.Discord/Areas/Users/Controllers/WarnsController.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Devscord.DiscordFramework.Commons;
+using Devscord.DiscordFramework.Commons.Exceptions;
+using Devscord.DiscordFramework.Commons.Extensions;
+using Devscord.DiscordFramework.Framework.Architecture.Controllers;
+using Devscord.DiscordFramework.Framework.Commands.Parsing.Models;
+using Devscord.DiscordFramework.Framework.Commands.Responses;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+using Devscord.DiscordFramework.Services;
+using Devscord.DiscordFramework.Services.Factories;
+using MongoDB.Bson;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Watchman.Discord.Areas.Protection.Services;
+using Watchman.Discord.Areas.Users.BotCommands;
+using Watchman.Discord.Areas.Users.BotCommands.Warns;
+using Watchman.DomainModel.Messages;
+using Watchman.DomainModel.Users;
+
+namespace Watchman.Discord.Areas.Protection.Controllers
+{
+    public class WarnsController : IController
+    {
+        private readonly MessagesServiceFactory _messagesServiceFactory;
+        private readonly UsersService _usersService;
+        private readonly DirectMessagesService _directMessagesService;
+        private readonly WarnsService _warnService;
+
+        public WarnsController(MessagesServiceFactory messagesServiceFactory, UsersService usersService, DirectMessagesService directMessagesService, WarnsService warnService)
+        {
+            this._messagesServiceFactory = messagesServiceFactory;
+            this._usersService = usersService;
+            this._directMessagesService = directMessagesService;
+            this._warnService = warnService;
+        }
+
+        [AdminCommand]
+        public async Task AddWarn(AddWarnCommand command, Contexts contexts)
+        {
+            var messageService = this._messagesServiceFactory.Create(contexts);
+            var mentionedUser = await this._usersService.GetUserByIdAsync(contexts.Server, command.User);
+            await this._warnService.AddWarnToUser(command, contexts, mentionedUser);
+            await messageService.SendResponse(x => x.UserHasBeenWarned(contexts.User.Name, mentionedUser.Name, command.Reason));
+        }
+
+        public async Task Warns(WarnsCommand command, Contexts contexts)
+        {
+            var messageService = _messagesServiceFactory.Create(contexts);
+            var mentionedUser = command.User == 0 ? 
+                    contexts.User : 
+                    await this._usersService.GetUserByIdAsync(contexts.Server, command.User);
+            var warns = this._warnService.GetWarns(command, contexts, mentionedUser, contexts.Server.Id);
+            if (warns == null) 
```

Comment:
niezbyt rozumiem tą logike

sprawdzasz czy warns == null
a później zwracasz UserNotFound

o co tutaj chodzi?
co jeśli użytkownik istnieje, ale akurat nie dostał żadnego ostrzeżenia?
albo co jak dostaniesz tutaj liste, z tym że z brakiem elementów

Reply:
funkcja GetWarns jest tak zrobiona, że jak zwraca null to znaczy że user nie istnieje, a jak istnieje ale nie ma żadnych warnów, to zwraca tylko jedną pozycję z informacją że nie ma warnów
![2 (2)](https://user-images.githubusercontent.com/47067171/89737741-c4702700-da73-11ea-9830-239d3e2d8a95.png)

Reply:
a nie wystarczy sprawdzać już wcześniej samo mentionedUser czy nie jest nullem?

Reply:
@gagyn w sumie tak